lib/socketpair.h

```diff
@@ -38,7 +38,14 @@
 #define wakeup_write     swrite
 #define wakeup_read      sread
 #define wakeup_close     sclose
-#define wakeup_create(p) Curl_socketpair(AF_UNIX, SOCK_STREAM, 0, p)
+
+#ifdef USE_UNIX_SOCKETS
+#define CURL_SOCKET_FAMILY AF_UNIX
+#else
+#define CURL_SOCKET_FAMILY AF_INET
+#endif
```

Comment:
If the native `socketpair()` function is used, it cannot be changed to AF_INET. And if the non-native implementation is used, the family is not used so might as well just be set to zero.

Also, no need to use `CURL_` prefix for a private define.

May I suggest:

```suggestion
#if defined(USE_UNIX_SOCKETS) && defined(HAVE_SOCKETPAIR)
#define SOCKET_FAMILY AF_UNIX
#elif !defined(USE_UNIX_SOCKETS) && !defined(HAVE_SOCKETPAIR)
#define SOCKET_FAMILY 0 /* not used */
#else
#error "unsupported unix domain and socketpair build combo"
#endif
```

Reply:
Thanks! I didn't know about that